packages/maath/src/misc.ts

```diff
@@ -3,6 +3,53 @@ import { doThreePointsMakeARight } from "./triangle";
 import type { TypedArray } from "./ctypes";
 import { matrixSum3 } from "./matrix";
 
+/**
+ * Clamps a value between a range.
+ */
+export function clamp([min, max]: [number, number], value: number) {
+  return Math.max(min, Math.min(max, value));
+}
+
+/**
+ * Converts degrees to radians.
+ */
+export function degToRad(degrees: number) {
+  return (degrees / 180) * Math.PI;
+}
+
+/**
+ * Converts radians to degrees.
+ */
+export function radToDeg(radians: number) {
+  return (radians * 180) / Math.PI;
+}
+
+/**
+ * Gets 3D coordinates from latitude and longitude along a sphere.
+ *
+ * @param latitude Target latitude in degrees.
+ * @param longitude Target longitude in degrees.
+ * @param size Target sphere's dimensions in pixels.
+ * @param radius Target sphere's radius.
+ */
+export function getSphericalCoordinates(
+  latitude: number,
+  longitude: number,
+  size: { width: number; height: number },
+  radius: number
```

Comment:
I'd like to make a common interface for common shapes, eg.

```ts
interface Sphere {
  radius: number,
  center: Vector3
}
```

and use it in sphere-related methods such as the random one. What do you think?

Reply:
Good idea. I think I'll wait on adding more 3D stuff until there's a common interface in place.